### PR TITLE
Allow skin components to have settings

### PR DESCRIPTION
**Pull request by peppy**
This adds a new toolbox in the skin editor to show skin component settings, as well as the groundwork for allowing them to exist in the first place.

Rather than adding settings to existing skinnables, I've created a sample `BigBlackBox` to show how this can work. I'd rather add settings to existing skinnables in separate PRs, but wanted something in this PR to allow for testing. Figure there's no harm in allowing users to test this as well while we expand the system further (and gracefully remove it in the future).

- [x] Depends on https://github.com/ppy/osu/pull/17218 for conflict avoidance
- [x] Depends on https://github.com/ppy/osu/pull/17217 for conflict avoidance

### CODE REVIEW

**Review by bdach — COMMENTED** [review]
General structure looks pretty sane to me

File: osu.Game/Screens/Play/HUD/SkinnableInfo.cs [hunk 1]
```diff
@@ -58,6 +64,14 @@ public SkinnableInfo(Drawable component)
             if (component is ISkinnableDrawable skinnable)
                 UsesFixedAnchor = skinnable.UsesFixedAnchor;
 
+            foreach (var (_, property) in component.GetSettingsSourceProperties())
+            {
+                var bindable = (IBindable)property.GetValue(component);
+
+                if (!bindable.IsDefault)
+                    Settings.Add(property.Name.Underscore(), ModUtils.GetSettingUnderlyingValue(bindable));
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game/Screens/Play/HUD/SkinnableInfo.cs:72]: Yeah for sure. Wasn't too sure where to put it, will make a new namespace.

**Review by frenzibyte — APPROVED** [review]
Looks pretty good.

**Review by bdach — APPROVED** [review]
Looks ok to me

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**